## Index of Multiple Deprivation etc - data exploration



The data file has been copied and modified slightly to make this easier to work with and is called example.csv. This is just an example investigation and you are welcome to explore and use any of the data files provided. 




In [1]:
!dir
#for windows dir and for lunix ls

 Volume in drive C has no label.
 Volume Serial Number is DA94-AA60

 Directory of C:\Users\AR\Downloads\DataScienceAssignment\2020J_TMA02

24/02/2021  09:18 pm    <DIR>          .
24/02/2021  09:18 pm    <DIR>          ..
24/02/2021  07:27 pm    <DIR>          .ipynb_checkpoints
24/02/2021  09:08 pm             5,848 tm351_tma02_readme.ipynb
24/02/2021  09:18 pm            12,101 TMA02_Q2_Load_and_Explore_Data.ipynb
27/11/2020  07:03 pm             9,786 TMA02_Q2_Load_and_Visualise_Data.ipynb
24/02/2021  07:29 pm               586 yourPI_TMA02_Question2b.ipynb
               4 File(s)         28,321 bytes
               3 Dir(s)  38,661,689,344 bytes free


In [2]:
!dir '../2020J_TMA02_data/Deprivation_Index'

Invalid switch - "2020J_TMA02_data".


In [3]:
!pwd

'pwd' is not recognized as an internal or external command,
operable program or batch file.


In [4]:
import pandas as pd

# furl = "Alldata2.csv"
furl = '../2020J_TMA02_data/Deprivation_Index/example.csv'

df = pd.read_csv(furl)
df.head()

LSOAcode2011 LSOAname2011 LocalAuthorityDistrictcode2013  \
0    E01031349    Adur 001A                      E07000223   
1    E01031350    Adur 001B                      E07000223   
2    E01031351    Adur 001C                      E07000223   
3    E01031352    Adur 001D                      E07000223   
4    E01031370    Adur 001E                      E07000223   

  LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
0                           Adur                         21,352   
1                           Adur                          8,864   
2                           Adur                         22,143   
3                           Adur                         17,252   
4                           Adur                         15,643   

   IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile EmploymentRank  \
0                                 7     18,992             6         19,305   
1                                 3      9,233             3          7,879   
2                                 7     24,539             8         23,389   
3                                 6     16,087             5         13,699   
4                                 5     17,918             6         13,322   

   EmploymentDecile EducationSkillsandTrainingRank   \
0                 6                          13,727   
1                 3                           6,969   
2                 8                          10,213   
3                 5                          10,468   
4                 5                           7,819   

   EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
0                                 5                             25,876   
1                                 3                              6,883   
2                                 4                             24,693   
3                                 4                             19,627   
4                                 3                             14,893   

   HealthDeprivationandDisabilityDecile CrimeRank  CrimeDecile   \
0                                     8    12,817             4   
1                                     3    12,781             4   
2                                     8     9,112             3   
3                                     6    16,127             5   
4                                     5    13,723             5   

  BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
0                            28,166                                    9   
1                            11,399                                    4   
2                            24,743                                    8   
3                            19,559                                    6   
4                            22,433                                    7   

  LivingEnvironmentRank   LivingEnvironmentDecile   
0                 18,367                         6  
1                 16,242                         5  
2                 22,299                         7  
3                 22,819                         7  
4                 18,191                         6

In [5]:
# Preview the values in one of the data files
!tail '../2020J_TMA02_data/Deprivation_Index/example.csv'

'tail' is not recognized as an internal or external command,
operable program or batch file.


This file that has been loaded has a number of columns. It is worth seeing what type of columns are available for the data



In [6]:
list(df.columns)

['LSOAcode2011',
 'LSOAname2011',
 'LocalAuthorityDistrictcode2013',
 'LocalAuthorityDistrictname2013',
 'IndexofMultipleDeprivationRank',
 'IndexofMultipleDeprivationDecile',
 'IncomeRank',
 'IncomeDecile',
 'EmploymentRank',
 'EmploymentDecile',
 'EducationSkillsandTrainingRank ',
 'EducationSkillsandTrainingDecile',
 'HealthDeprivationandDisabilityRank',
 'HealthDeprivationandDisabilityDecile',
 'CrimeRank',
 'CrimeDecile ',
 'BarrierstoHousingandServicesRank ',
 'BarrierstoHousingandServicesDecile ',
 'LivingEnvironmentRank ',
 'LivingEnvironmentDecile ']

In [7]:
ListofNames = df['LocalAuthorityDistrictname2013'].unique()
ListofNames

array(['Adur', 'Allerdale', 'Amber Valley', 'Arun', 'Ashfield', 'Ashford',
       'Aylesbury Vale', 'Babergh', 'Barking and Dagenham', 'Barnet',
       'Barnsley', 'Barrow-in-Furness', 'Basildon',
       'Basingstoke and Deane', 'Bassetlaw',
       'Bath and North East Somerset', 'Bedford', 'Bexley', 'Birmingham',
       'Blaby', 'Blackburn with Darwen', 'Blackpool', 'Bolsover',
       'Bolton', 'Boston', 'Bournemouth', 'Bracknell Forest', 'Bradford',
       'Braintree', 'Breckland', 'Brent', 'Brentwood',
       'Brighton and Hove', 'Bristol, City of', 'Broadland', 'Bromley',
       'Bromsgrove', 'Broxbourne', 'Broxtowe', 'Burnley', 'Bury',
       'Calderdale', 'Cambridge', 'Camden', 'Cannock Chase', 'Canterbury',
       'Carlisle', 'Castle Point', 'Central Bedfordshire', 'Charnwood',
       'Chelmsford', 'Cheltenham', 'Cherwell', 'Cheshire East',
       'Cheshire West and Chester', 'Chesterfield', 'Chichester',
       'Chiltern', 'Chorley', 'Christchurch', 'City of London',
       'Co

the list can be used to find the areas that you are interested in 

In [8]:
df[df.LocalAuthorityDistrictname2013 == 'Bradford'].head()

LSOAcode2011   LSOAname2011 LocalAuthorityDistrictcode2013  \
3095    E01010646  Bradford 001A                      E08000032   
3096    E01010647  Bradford 001B                      E08000032   
3097    E01010648  Bradford 001C                      E08000032   
3098    E01010692  Bradford 001D                      E08000032   
3099    E01010691  Bradford 002A                      E08000032   

     LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
3095                       Bradford                         30,426   
3096                       Bradford                         26,378   
3097                       Bradford                         29,500   
3098                       Bradford                         29,485   
3099                       Bradford                         32,725   

      IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile  \
3095                                10     27,564             9   
3096                                 9     20,319             7   
3097                                 9     27,109             9   
3098                                 9     30,820            10   
3099                                10     31,852            10   

     EmploymentRank  EmploymentDecile EducationSkillsandTrainingRank   \
3095         29,855                10                          31,470   
3096         22,965                 7                          28,414   
3097         27,437                 9                          30,579   
3098         26,796                 9                          32,677   
3099         32,530                10                          32,786   

      EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
3095                                10                             22,927   
3096                                 9                             12,912   
3097                                10                             19,905   
3098                                10                             22,549   
3099                                10                             27,915   

      HealthDeprivationandDisabilityDecile CrimeRank  CrimeDecile   \
3095                                     7    32,447            10   
3096                                     4    31,074            10   
3097                                     7    32,574            10   
3098                                     7    23,108             8   
3099                                     9    29,647            10   

     BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
3095                            25,639                                    8   
3096                            31,616                                   10   
3097                            19,208                                    6   
3098                            20,126                                    7   
3099                            29,258                                    9   

     LivingEnvironmentRank   LivingEnvironmentDecile   
3095                 14,518                         5  
3096                 25,406                         8  
3097                 21,196                         7  
3098                 16,628                         6  
3099                 25,899                         8

The local authority district code might be interesting later or the area code.

In [9]:
df['LocalAuthorityDistrictcode2013'].unique()

array(['E07000223', 'E07000026', 'E07000032', 'E07000224', 'E07000170',
       'E07000105', 'E07000004', 'E07000200', 'E09000002', 'E09000003',
       'E08000016', 'E07000027', 'E07000066', 'E07000084', 'E07000171',
       'E06000022', 'E06000055', 'E09000004', 'E08000025', 'E07000129',
       'E06000008', 'E06000009', 'E07000033', 'E08000001', 'E07000136',
       'E06000028', 'E06000036', 'E08000032', 'E07000067', 'E07000143',
       'E09000005', 'E07000068', 'E06000043', 'E06000023', 'E07000144',
       'E09000006', 'E07000234', 'E07000095', 'E07000172', 'E07000117',
       'E08000002', 'E08000033', 'E07000008', 'E09000007', 'E07000192',
       'E07000106', 'E07000028', 'E07000069', 'E06000056', 'E07000130',
       'E07000070', 'E07000078', 'E07000177', 'E06000049', 'E06000050',
       'E07000034', 'E07000225', 'E07000005', 'E07000118', 'E07000048',
       'E09000001', 'E07000071', 'E07000029', 'E07000150', 'E06000052',
       'E07000079', 'E06000047', 'E08000026', 'E07000163', 'E070

## Once you have an area of interest you might want to filter on that area. Here is an example using Bradford

In [10]:
df.query('LocalAuthorityDistrictname2013 =="Bradford"')

LSOAcode2011   LSOAname2011 LocalAuthorityDistrictcode2013  \
3095    E01010646  Bradford 001A                      E08000032   
3096    E01010647  Bradford 001B                      E08000032   
3097    E01010648  Bradford 001C                      E08000032   
3098    E01010692  Bradford 001D                      E08000032   
3099    E01010691  Bradford 002A                      E08000032   
...           ...            ...                            ...   
3400    E01010869  Bradford 061B                      E08000032   
3401    E01010870  Bradford 061C                      E08000032   
3402    E01010871  Bradford 061D                      E08000032   
3403    E01010872  Bradford 061E                      E08000032   
3404    E01010873  Bradford 061F                      E08000032   

     LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
3095                       Bradford                         30,426   
3096                       Bradford                         26,378   
3097                       Bradford                         29,500   
3098                       Bradford                         29,485   
3099                       Bradford                         32,725   
...                             ...                            ...   
3400                       Bradford                          6,501   
3401                       Bradford                         17,084   
3402                       Bradford                          9,227   
3403                       Bradford                          2,927   
3404                       Bradford                         14,426   

      IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile  \
3095                                10     27,564             9   
3096                                 9     20,319             7   
3097                                 9     27,109             9   
3098                                 9     30,820            10   
3099                                10     31,852            10   
...                                ...        ...           ...   
3400                                 2      7,226             3   
3401                                 6     18,086             6   
3402                                 3     11,842             4   
3403                                 1      3,490             2   
3404                                 5     16,987             6   

     EmploymentRank  EmploymentDecile EducationSkillsandTrainingRank   \
3095         29,855                10                          31,470   
3096         22,965                 7                          28,414   
3097         27,437                 9                          30,579   
3098         26,796                 9                          32,677   
3099         32,530                10                          32,786   
...             ...               ...                             ...   
3400          6,851                 3                           6,437   
3401         17,443                 6                          18,320   
3402         10,403                 4                           9,361   
3403          2,625                 1                           2,082   
3404         15,909                 5                          10,983   

      EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
3095                                10                             22,927   
3096                                 9                             12,912   
3097                                10                             19,905   
3098                                10                             22,549   
3099                                10                             27,915   
...                                ...                                ...   
3400                                 2                              5,613   
3401                                 6                       

There are over 57 columns - it would be good to have a list of what type of data is being held - remember list(df.columns)

In [11]:
list(df.columns)

['LSOAcode2011',
 'LSOAname2011',
 'LocalAuthorityDistrictcode2013',
 'LocalAuthorityDistrictname2013',
 'IndexofMultipleDeprivationRank',
 'IndexofMultipleDeprivationDecile',
 'IncomeRank',
 'IncomeDecile',
 'EmploymentRank',
 'EmploymentDecile',
 'EducationSkillsandTrainingRank ',
 'EducationSkillsandTrainingDecile',
 'HealthDeprivationandDisabilityRank',
 'HealthDeprivationandDisabilityDecile',
 'CrimeRank',
 'CrimeDecile ',
 'BarrierstoHousingandServicesRank ',
 'BarrierstoHousingandServicesDecile ',
 'LivingEnvironmentRank ',
 'LivingEnvironmentDecile ']

The reports provided explain about how the numerical values - how they are calculated and their significance. 
The lower the value e.g.  the number 1 is the most deprived in the Index of Multiple Deprivation factors.

This example looks at extracting for Bradford the those areas with a value of 1


In [12]:
df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile == 1')

LSOAcode2011   LSOAname2011 LocalAuthorityDistrictcode2013  \
3127    E01010719  Bradford 007A                      E08000032   
3131    E01010700  Bradford 008A                      E08000032   
3132    E01010701  Bradford 008B                      E08000032   
3133    E01010702  Bradford 008C                      E08000032   
3137    E01010710  Bradford 008G                      E08000032   
...           ...            ...                            ...   
3387    E01010845  Bradford 059B                      E08000032   
3389    E01010850  Bradford 059D                      E08000032   
3393    E01010874  Bradford 059H                      E08000032   
3399    E01010868  Bradford 061A                      E08000032   
3403    E01010872  Bradford 061E                      E08000032   

     LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
3127                       Bradford                          2,832   
3131                       Bradford                          1,772   
3132                       Bradford                            691   
3133                       Bradford                          2,777   
3137                       Bradford                            284   
...                             ...                            ...   
3387                       Bradford                          1,425   
3389                       Bradford                            547   
3393                       Bradford                          2,482   
3399                       Bradford                          1,923   
3403                       Bradford                          2,927   

      IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile  \
3127                                 1      3,770             2   
3131                                 1      2,165             1   
3132                                 1        757             1   
3133                                 1      5,077             2   
3137                                 1      1,074             1   
...                                ...        ...           ...   
3387                                 1      1,175             1   
3389                                 1        830             1   
3393                                 1      3,170             1   
3399                                 1      1,818             1   
3403                                 1      3,490             2   

     EmploymentRank  EmploymentDecile EducationSkillsandTrainingRank   \
3127          2,649                 1                           2,018   
3131          3,306                 2                             818   
3132          1,706                 1                             454   
3133          4,668                 2                             924   
3137            713                 1                             911   
...             ...               ...                             ...   
3387          2,764                 1                             893   
3389          1,044                 1                              86   
3393          2,905                 1                           2,131   
3399          2,129                 1                           3,006   
3403          2,625                 1                           2,082   

      EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
3127                                 1                              4,944   
3131                                 1                              6,028   
3132                                 1                              1,250   
3133                                 1                              4,756   
3137                                 1                              1,483   
...                                ...                                ...   
3387                                 1                              1,808   
3389                                 1                       

Let's look a bit more the data in general - linked to the Index of Multiple Deprivation is the rank value. 

In [13]:
df_wide = df.pivot(index= 'IndexofMultipleDeprivationDecile', columns='LSOAname2011', values='IndexofMultipleDeprivationRank')
df_wide.head()

LSOAname2011                     Adur 001A Adur 001B Adur 001C Adur 001D  \
IndexofMultipleDeprivationDecile                                           
1                                      NaN       NaN       NaN       NaN   
2                                      NaN       NaN       NaN       NaN   
3                                      NaN     8,864       NaN       NaN   
4                                      NaN       NaN       NaN       NaN   
5                                      NaN       NaN       NaN       NaN   

LSOAname2011                     Adur 001E Adur 001F Adur 002A Adur 002B  \
IndexofMultipleDeprivationDecile                                           
1                                      NaN       NaN       NaN       NaN   
2                                      NaN       NaN       NaN       NaN   
3                                      NaN       NaN       NaN       NaN   
4                                      NaN       NaN       NaN       NaN   
5                                   15,643       NaN       NaN       NaN   

LSOAname2011                     Adur 002C Adur 002D  ... York 022F York 023A  \
IndexofMultipleDeprivationDecile                      ...                       
1                                      NaN       NaN  ...       NaN       NaN   
2                                      NaN       NaN  ...       NaN       NaN   
3                                      NaN       NaN  ...       NaN       NaN   
4                                      NaN       NaN  ...       NaN       NaN   
5                                      NaN       NaN  ...       NaN       NaN   

LSOAname2011                     York 023B York 023C York 024A York 024B  \
IndexofMultipleDeprivationDecile                                           
1                                      NaN       NaN       NaN       NaN   
2                                      NaN       NaN       NaN       NaN   
3                                      NaN       NaN       NaN       NaN   
4                                      NaN       NaN       NaN       NaN   
5                                      NaN       NaN       NaN       NaN   

LSOAname2011                     York 024C York 024D York 024E York 024F  
IndexofMultipleDeprivationDecile                                          
1                                      NaN       NaN       NaN       NaN  
2                                      NaN       NaN       NaN       NaN  
3                                      NaN       NaN       NaN       NaN  
4                                      NaN       NaN       NaN       NaN  
5                                      NaN       NaN       NaN       NaN  

[5 rows x 32844 columns]

A quick glance (although a better filter and table could be used here) - shows that areas e.g. Adur and York rank value above 4000 puts in the deprivation value above 2 - this is just glancing at the data. Other filters and queries can be developed to check this. Data could be organised to investigate to see for example which factors might at the most deprived level are impacting the chosen area.

If we review the document The English Index of Multiple Deprivation
(IMD) 2015 – Guidance

see page 2 "Small area X is ranked 5,000 out of 32,844 small-areas in England, where 1 is the most deprived. This means that small area X is amongst the 20 per cent most deprived small areas in the country."

See page 10 of The English Indices of Deprivation 2015 – Frequently Asked Questions (FAQs) to see the table of ranking

Quick look at the data to see if anything interesting 

In [14]:
df_wide.describe()

LSOAname2011 Adur 001A Adur 001B Adur 001C Adur 001D Adur 001E Adur 001F  \
count                1         1         1         1         1         1   
unique               1         1         1         1         1         1   
top             21,352     8,864    22,143    17,252    15,643    21,176   
freq                 1         1         1         1         1         1   

LSOAname2011 Adur 002A Adur 002B Adur 002C Adur 002D  ... York 022F York 023A  \
count                1         1         1         1  ...         1         1   
unique               1         1         1         1  ...         1         1   
top             27,934    28,249    27,569    18,352  ...    19,988    28,286   
freq                 1         1         1         1  ...         1         1   

LSOAname2011 York 023B York 023C York 024A York 024B York 024C York 024D  \
count                1         1         1         1         1         1   
unique               1         1         1         1         1         1   
top             29,849    25,852    32,752    29,241    25,821    32,637   
freq                 1         1         1         1         1         1   

LSOAname2011 York 024E York 024F  
count                1         1  
unique               1         1  
top             32,799    31,273  
freq                 1         1  

[4 rows x 32844 columns]

Let's focus on Bradford filter for IMD of 1 and 2

In [15]:
tmp = df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile < 3')
tmp1_wide = tmp.pivot(index= 'IndexofMultipleDeprivationDecile', columns='LSOAname2011', values='IndexofMultipleDeprivationRank')
tmp1_wide.head()

LSOAname2011                     Bradford 007A Bradford 007C Bradford 008A  \
IndexofMultipleDeprivationDecile                                             
1                                        2,832           NaN         1,772   
2                                          NaN         3,893           NaN   

LSOAname2011                     Bradford 008B Bradford 008C Bradford 008G  \
IndexofMultipleDeprivationDecile                                             
1                                          691         2,777           284   
2                                          NaN           NaN           NaN   

LSOAname2011                     Bradford 009A Bradford 009B Bradford 009C  \
IndexofMultipleDeprivationDecile                                             
1                                        2,917         2,088         1,541   
2                                          NaN           NaN           NaN   

LSOAname2011                     Bradford 009D  ... Bradford 056D  \
IndexofMultipleDeprivationDecile                ...                 
1                                        1,214  ...           NaN   
2                                          NaN  ...         4,387   

LSOAname2011                     Bradford 057A Bradford 057B Bradford 057D  \
IndexofMultipleDeprivationDecile                                             
1                                        2,988           NaN         3,130   
2                                          NaN         5,717           NaN   

LSOAname2011                     Bradford 059B Bradford 059D Bradford 059H  \
IndexofMultipleDeprivationDecile                                             
1                                        1,425           547         2,482   
2                                          NaN           NaN           NaN   

LSOAname2011                     Bradford 061A Bradford 061B Bradford 061E  
IndexofMultipleDeprivationDecile                                            
1                                        1,923           NaN         2,927  
2                                          NaN         6,501           NaN  

[2 rows x 131 columns]

From initial analysis of the IMD of there are 131 areas in Bradford that fall into the IMD of 1 and 2. We have over a number of columns that would be interesting to investigate.

In [16]:
list(df.columns)

['LSOAcode2011',
 'LSOAname2011',
 'LocalAuthorityDistrictcode2013',
 'LocalAuthorityDistrictname2013',
 'IndexofMultipleDeprivationRank',
 'IndexofMultipleDeprivationDecile',
 'IncomeRank',
 'IncomeDecile',
 'EmploymentRank',
 'EmploymentDecile',
 'EducationSkillsandTrainingRank ',
 'EducationSkillsandTrainingDecile',
 'HealthDeprivationandDisabilityRank',
 'HealthDeprivationandDisabilityDecile',
 'CrimeRank',
 'CrimeDecile ',
 'BarrierstoHousingandServicesRank ',
 'BarrierstoHousingandServicesDecile ',
 'LivingEnvironmentRank ',
 'LivingEnvironmentDecile ']

You might want to explore specific data areas for an area. 
Filter the data further. Some examples are provided - this you have already seen - slightly different IMD parameter

In [17]:
df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile > 3')

LSOAcode2011   LSOAname2011 LocalAuthorityDistrictcode2013  \
3095    E01010646  Bradford 001A                      E08000032   
3096    E01010647  Bradford 001B                      E08000032   
3097    E01010648  Bradford 001C                      E08000032   
3098    E01010692  Bradford 001D                      E08000032   
3099    E01010691  Bradford 002A                      E08000032   
...           ...            ...                            ...   
3395    E01010750  Bradford 060B                      E08000032   
3397    E01010865  Bradford 060D                      E08000032   
3398    E01010866  Bradford 060E                      E08000032   
3401    E01010870  Bradford 061C                      E08000032   
3404    E01010873  Bradford 061F                      E08000032   

     LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
3095                       Bradford                         30,426   
3096                       Bradford                         26,378   
3097                       Bradford                         29,500   
3098                       Bradford                         29,485   
3099                       Bradford                         32,725   
...                             ...                            ...   
3395                       Bradford                         16,641   
3397                       Bradford                         13,918   
3398                       Bradford                         12,844   
3401                       Bradford                         17,084   
3404                       Bradford                         14,426   

      IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile  \
3095                                10     27,564             9   
3096                                 9     20,319             7   
3097                                 9     27,109             9   
3098                                 9     30,820            10   
3099                                10     31,852            10   
...                                ...        ...           ...   
3395                                 6     17,521             6   
3397                                 5     17,887             6   
3398                                 4     14,150             5   
3401                                 6     18,086             6   
3404                                 5     16,987             6   

     EmploymentRank  EmploymentDecile EducationSkillsandTrainingRank   \
3095         29,855                10                          31,470   
3096         22,965                 7                          28,414   
3097         27,437                 9                          30,579   
3098         26,796                 9                          32,677   
3099         32,530                10                          32,786   
...             ...               ...                             ...   
3395         17,678                 6                          12,168   
3397         14,299                 5                          10,140   
3398         14,791                 5                           9,777   
3401         17,443                 6                          18,320   
3404         15,909                 5                          10,983   

      EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
3095                                10                             22,927   
3096                                 9                             12,912   
3097                                10                             19,905   
3098                                10                             22,549   
3099                                10                             27,915   
...                                ...                                ...   
3395                                 4                             12,168   
3397                                 4                       

let's look at the overall data for Bradford

In [18]:
tmp2 = df.query('LocalAuthorityDistrictname2013 =="Bradford"')
tmp3_wide = tmp2.pivot(index= 'LSOAname2011', columns='IndexofMultipleDeprivationDecile', values='IndexofMultipleDeprivationRank')
tmp3_wide.head(300)

IndexofMultipleDeprivationDecile     1    2      3    4       5    6    7   \
LSOAname2011                                                                 
Bradford 001A                       NaN  NaN    NaN  NaN     NaN  NaN  NaN   
Bradford 001B                       NaN  NaN    NaN  NaN     NaN  NaN  NaN   
Bradford 001C                       NaN  NaN    NaN  NaN     NaN  NaN  NaN   
Bradford 001D                       NaN  NaN    NaN  NaN     NaN  NaN  NaN   
Bradford 002A                       NaN  NaN    NaN  NaN     NaN  NaN  NaN   
...                                 ...  ...    ...  ...     ...  ...  ...   
Bradford 059E                       NaN  NaN    NaN  NaN  15,540  NaN  NaN   
Bradford 059F                       NaN  NaN  6,999  NaN     NaN  NaN  NaN   
Bradford 059G                       NaN  NaN  8,887  NaN     NaN  NaN  NaN   
Bradford 059H                     2,482  NaN    NaN  NaN     NaN  NaN  NaN   
Bradford 060A                       NaN  NaN    NaN  NaN  14,403  NaN  NaN   

IndexofMultipleDeprivationDecile   8       9       10  
LSOAname2011                                           
Bradford 001A                     NaN     NaN  30,426  
Bradford 001B                     NaN  26,378     NaN  
Bradford 001C                     NaN  29,500     NaN  
Bradford 001D                     NaN  29,485     NaN  
Bradford 002A                     NaN     NaN  32,725  
...                               ...     ...     ...  
Bradford 059E                     NaN     NaN     NaN  
Bradford 059F                     NaN     NaN     NaN  
Bradford 059G                     NaN     NaN     NaN  
Bradford 059H                     NaN     NaN     NaN  
Bradford 060A                     NaN     NaN     NaN  

[300 rows x 10 columns]

In [19]:
tmp3_wide.describe

<bound method NDFrame.describe of IndexofMultipleDeprivationDecile     1      2      3    4       5       6   \
LSOAname2011                                                                 
Bradford 001A                       NaN    NaN    NaN  NaN     NaN     NaN   
Bradford 001B                       NaN    NaN    NaN  NaN     NaN     NaN   
Bradford 001C                       NaN    NaN    NaN  NaN     NaN     NaN   
Bradford 001D                       NaN    NaN    NaN  NaN     NaN     NaN   
Bradford 002A                       NaN    NaN    NaN  NaN     NaN     NaN   
...                                 ...    ...    ...  ...     ...     ...   
Bradford 061B                       NaN  6,501    NaN  NaN     NaN     NaN   
Bradford 061C                       NaN    NaN    NaN  NaN     NaN  17,084   
Bradford 061D                       NaN    NaN  9,227  NaN     NaN     NaN   
Bradford 061E                     2,927    NaN    NaN  NaN     NaN     NaN   
Bradford 061F                 

Filter on more specific data e.g. Education Skills

In [20]:
tmp3_wide = tmp2.pivot(index= 'LSOAname2011', columns= 'EducationSkillsandTrainingDecile', values='EducationSkillsandTrainingRank ')
tmp3_wide.head(50)

EducationSkillsandTrainingDecile     1      2      3       4       5       6   \
LSOAname2011                                                                    
Bradford 001A                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 001B                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 001C                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 001D                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002A                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002B                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002C                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002D                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002E                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002F                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002G                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 002H                       NaN    NaN    NaN     NaN  15,661     NaN   
Bradford 003A                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 003B                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 003C                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 003D                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004A                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004B                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004C                       NaN    NaN    NaN     NaN     NaN  19,198   
Bradford 004D                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004E                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004F                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004G                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 004H                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 005A                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 005B                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 005C                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 005D                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 006A                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 006B                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 006C                       NaN    NaN    NaN  12,054     NaN     NaN   
Bradford 006D                       NaN    NaN    NaN     NaN     NaN     NaN   
Bradford 007A                     2,018    NaN    NaN     NaN     NaN     NaN   
Bradford 007B                       NaN    NaN    NaN     NaN     NaN  17,381   
Bradford 007C                     2,863    NaN    NaN     NaN     NaN     NaN   
Bradford 007D                       NaN    NaN    NaN     NaN  13,413     NaN   
Bradford 008A                       818    NaN    NaN     NaN     NaN     NaN   
Bradford 008B                       454    NaN    NaN     NaN     NaN     NaN   
Bradford 008C                       924    NaN    NaN     NaN     NaN     NaN   
Bradford 008D                       NaN    NaN  7,001     NaN     NaN     NaN   
Bradford 008E                       NaN    NaN    NaN  10,170     NaN     NaN   
Bradford 008F                       NaN    NaN    NaN     NaN  13,388     NaN   
Bradford 008G                       911    NaN    NaN     NaN     NaN     NaN   
Bradford 009A                     1,700    NaN    NaN     NaN     NaN     NaN   
Bradford 009B                       830    NaN    NaN     NaN     NaN     NaN   
Bradford 009C                       771    NaN    NaN     NaN     NaN     NaN   
Bradford 009D                       403    NaN    NaN     NaN     NaN     NaN   
Bradford 009E                  

Data can be filtered, investigated and sumerised on the different columns. After exploring this and areas of interest then the students can start to decide on what aspects to investigate and compare.


Filter further on skills analysis for those Adults in the most deprived areas of Bradford. e.g. IMD = 1

In [21]:
tmp2 = df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile == 1')
tmp3_wide = tmp2.pivot(index= 'LSOAname2011',columns = 'EducationSkillsandTrainingDecile', values='EducationSkillsandTrainingRank ')
tmp3_wide.head(101)

EducationSkillsandTrainingDecile      1    2
LSOAname2011                                
Bradford 007A                     2,018  NaN
Bradford 008A                       818  NaN
Bradford 008B                       454  NaN
Bradford 008C                       924  NaN
Bradford 008G                       911  NaN
...                                 ...  ...
Bradford 059B                       893  NaN
Bradford 059D                        86  NaN
Bradford 059H                     2,131  NaN
Bradford 061A                     3,006  NaN
Bradford 061E                     2,082  NaN

[101 rows x 2 columns]